libbpf-rs/tests/test_tc.rs

```diff
@@ -0,0 +1,307 @@
+use serial_test::serial;
+
+mod test;
+use test::{bump_rlimit_mlock, get_test_object};
+
+use nix::errno::Errno::{EINVAL, ENOENT};
+
+use libbpf_rs::{
+    Error, Object, Result, TcHook, TcHookBuilder, TC_CUSTOM, TC_EGRESS, TC_H_CLSACT,
+    TC_H_MIN_EGRESS, TC_H_MIN_INGRESS, TC_INGRESS,
+};
+// do all TC tests on the lo network interface
+const LO_IFINDEX: i32 = 1;
+
+fn clear_clsact(fd: i32) -> Result<()> {
+    // Ensure clean clsact tc qdisc
+    let mut destroyer = TcHook::new(fd);
+    destroyer
+        .ifindex(LO_IFINDEX)
+        .attach_point(TC_EGRESS | TC_INGRESS);
+
+    let res = destroyer.destroy();
+    if let Err(Error::System(err)) = res {
+        if err != -(ENOENT as i32) && err != -(EINVAL as i32) {
```

Comment:
Should -EINVAL  be ignored or should that be treated  as a real error?

Reply:
Gotcha. If I'm understanding this correctly, what you're saying is that -EINVAL is not an error since the ingress queue handle is 0 in the success case (where the ingress queue is created and handle is kzalloc-ed). Is my understanding correct?